### PR TITLE
apparmor: create an apparmor namespace for each container

### PR DESCRIPTION
**Pull request by tych0**
Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>

### CODE REVIEW

File: lxd/apparmor.go [hunk 1]
```diff
@@ -317,12 +370,13 @@ func runApparmor(command string, c container) error {
...
 
-	cmd := exec.Command("apparmor_parser", []string{
+	args := []string{
 		fmt.Sprintf("-%sWL", command),
 		path.Join(aaPath, "cache"),
 		path.Join(aaPath, "profiles", AAProfileShort(c)),
-	}...)
+	}
 
+	cmd := exec.Command("apparmor_parser", args...)
```
Comment (stgraber) [inline lxd/apparmor.go]: Hmm, what am I missing here? looks identical to the old code

File: lxd/daemon.go [hunk 2]
```diff
@@ -626,6 +627,48 @@ func (d *Daemon) Init() error {
...
+			minor, err := strconv.Atoi(parts[1])
+			if err != nil {
+				shared.LogWarn("unknown apparmor domain version", log.Ctx{"version": content})
+				return false
+			}
+
+			return major >= 1 && minor >= 1
+		}
+
+		aaStacking = canStack()
```
Comment (stgraber) [inline lxd/daemon.go]: Hmm, why the separate function?

File: lxd/apparmor.go (hunk 1 above)
Comment (tych0) [inline lxd/apparmor.go]: It probably is. I suppose it's left over from some previous incantation of how I was doing this.

File: lxd/daemon.go (hunk 2 above)
Comment (tych0) [inline lxd/daemon.go]: It flattens the logic, instead of having a bunch of nested ifs.

**Pull request merged**